framework/include/utils/VectorComponentFunctor.h

```diff
@@ -0,0 +1,91 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseFunctor.h"
+
+/**
+ * This is essentially a forwarding functor that forwards the spatial and temporal evaluation
+ * arguments to the parent vector functor and then returns the result indexed at a given component.
+ */
+template <typename T>
```

Comment:
A `FunctorEnvelope` has the same return type as the `FunctorBase` that it wraps. I don't see a need for a differently named `FunctorEnvelope` based on return type since it's templated on the return type.

There's not yet a place where a `VectorComponentFunctor` is wrapped by a `FunctorEnvelope`. It's currently only used "internally" by the vector overload of `greenGaussGradient`.